packages/backend/src/services/SpaceService/SpaceService.test.ts

```diff
@@ -0,0 +1,865 @@
+                {
+                    name: 'can still update public space when group has viewer role (no downgrade)',
+                    user: { projectRole: ProjectMemberRole.EDITOR },
+                    space: { isPrivate: false },
+                    access: { groupSpaceRole: SpaceMemberRole.VIEWER },
+                    action: 'update',
+                    expectedResult: true,
+                    contentType: 'Space',
+                },
+                // Mixed role cases (group takes priority)
+                {
+                    name: 'group viewer role overrides direct editor role (downgrade)',
+                    user: { projectRole: ProjectMemberRole.EDITOR },
+                    space: { isPrivate: true },
+                    access: {
+                        spaceRole: SpaceMemberRole.EDITOR,
+                        groupSpaceRole: SpaceMemberRole.VIEWER,
+                    },
+                    action: 'update',
+                    expectedResult: false,
+                    contentType: 'Space',
+                },
+                // Dashboard specific cases
+                {
+                    name: 'can update dashboard in public space by default',
+                    user: { projectRole: ProjectMemberRole.EDITOR },
+                    space: { isPrivate: false },
+                    access: {},
+                    action: 'update',
+                    expectedResult: true,
+                    contentType: 'Dashboard',
+                },
+                {
+                    name: 'cannot update dashboard when space role is viewer',
+                    user: { projectRole: ProjectMemberRole.EDITOR },
+                    space: { isPrivate: true },
+                    access: { spaceRole: SpaceMemberRole.VIEWER },
+                    action: 'update',
+                    expectedResult: false,
+                    contentType: 'Dashboard',
+                },
+                {
+                    name: 'cannot update dashboard when group space role is viewer',
+                    user: { projectRole: ProjectMemberRole.EDITOR },
+                    space: { isPrivate: true },
+                    access: { groupSpaceRole: SpaceMemberRole.VIEWER },
+                    action: 'update',
+                    expectedResult: false,
+                    contentType: 'Dashboard',
+                },
+
+                // Promoting
+                // This fails because editors have manage permission on dashboards, so they can do any action (including promote)
+                // The intended behaviour is that only developers can promote dashboards
+                // {
+                //     name: 'cannot promote dashboard in public space',
+                //     user: { projectRole: ProjectMemberRole.EDITOR },
+                //     space: { isPrivate: false },
+                //     access: {},
+                //     action: 'promote',
+                //     expectedResult: false,
```

Comment:
Should we enable this test with `expectedResult: true,` ? 

Do we have a ticket for fixing this ?